File: Baekjoon/문제풀이/쇠막대기/yuri.py
s = input().strip()
stack = []
count = 0

for i in range(len(s)):
    if s[i] == '(':
        stack.append('(')
    else:
        if s[i - 1] == '(':
            stack.pop()
            count += len(stack)
        else:
            stack.pop()
            count += 1

print(count)

File: Baekjoon/문제풀이/쇠막대기/yuri.java
import java.util.Scanner;
import java.util.Stack;

public class yuri {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        Stack<Character> stack = new Stack<>();
        int count = 0;

        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);

            if (c == '(') {
                stack.push(c);
            } else {
                if (input.charAt(i - 1) == '(') {
                    stack.pop();
                    count += stack.size();
                } else {
                    stack.pop();
                    count++;
                }
            }
        }

        System.out.println(count);
    }
}
